lib/rest/Twilio.js

```diff
@@ -271,6 +268,40 @@ Twilio.prototype.request = function request(opts) {
   });
 };
 
+/* jshint ignore:start */
+/**
+ * Adds a region and/or edge to a given hostname
+ *
+ * @function getHostname
+ * @memberof Twilio#
+ *
+ * @param {string} hostname - A URI hostname (e.g. api.twilio.com)
+ * @param {string} targetEdge - The targeted edge location (e.g. sydney)
+ * @param {string} targetRegion - The targeted region location (e.g. au1)
+ */
+/* jshint ignore:end */
+Twilio.prototype.getHostname = function getHostname(hostname, targetEdge,
+                                                     targetRegion) {
+  const domain = hostname.split('.').slice(-2).join('.');
+  const defaultRegion = 'us1';
+
+  const prefix = hostname.split('.' + domain)[0];
+  let [product, edge, region] = prefix.split('.');
+
+  targetRegion = targetRegion || (targetEdge && defaultRegion);
+  if (!targetRegion) {
```

Comment:
Should move this check to before the hostname parsing logic. No need to parse the hostname if we're not modifying it.